lib/sequelize.js

```diff
@@ -227,6 +231,12 @@ class Sequelize {
 
     this.dialect = new Dialect(this);
     this.dialect.QueryGenerator.typeValidation = options.typeValidation;
+    if (this.options.operatorsAliases) {
+      this.dialect.QueryGenerator.setOperatorsAliases(this.options.operatorsAliases);
+    } else {
+      Utils.deprecate('Limiting available operators aliases offers improved security and will be forced in future versions. check: http://docs.sequelizejs.com/manual/tutorial/querying.html#operators');
```

Comment:
This should only be shown for FIRST time when user is actually trying to use legacy alias in code, otherwise I will get annoyed even if I am not using string based operators.

OR may be its missing `else if ( this.options.allowLegacyOperators )` ????

Message should say `String based operators are now deprecated. Please use Symbol based operators for better security, read more at http://docs.sequelizejs.com/manual/tutorial/querying.html#operators`

Reply:
The problem is if you don't select your own aliases (or completely disable them) they are available (for backward compatibility) so most users are not even aware of the aliases available and are not using many of them (legacy ones weren't even documented) but they can still be injected even if they are not normally used.
I think showing this warning unless the user sets his own operators using the `operatorsAliases` is the best option to make sure users are aware of the issue.
I tried to make it clear in the documentation how to use `operatorsAliases` and dismiss this warning but maybe it can be made clearer

Reply:
I liked the idea of default enabled flag, which will all load legacy alias, so this wont be a breaking change.

If they want better security and no deprecation warning they should just set that flag to `false`, Please remove user supplied `operatorsAliases` altogether.

Reply:
That's how I tried to implement it the first time.
But there are a few issues with doing it that way - 

To get a fully backward supported and secure options you need two flags one for legacy aliases (which are not prefixed undocumented and probably 99% of the users we have are not using them and so hopefully would disable them) and another for the $ prefixed operators which all users are probably using and we should hope they'll switch to symbols after this is released.

since we have two flags we we'll need two warnings -
1. Deprecation warning if legacy operators are enabled (even explicitly) since they will be removed any way in future versions
2. Deprecation warning for string operators not being explicitly enabled (I assume those will still be available in future version but won't be enabled by default)

Now I'm ending up setting two flags to hide all deprecation warnings of a single feature.
Users are not really aware of what they are enabling or disabling. 
They might only '..' alias and since it would fail enable legacy operators but then they'll be forced to also enable `@>`, `gt` and all the other legacy one.

Giving the the `operatorsAliases` takes control back to the user. we don't want to set those aliases for you cause you need to be aware for them. you want '..' to equal the between operator sure you can do it. want '||' to equal the or operator yea sure. But it is up to you to define those and up to you to sanitize them. Now every user knows exactly which strings will be translated into operators cause he is the one explicitly defined them.

So if my API allows client to send requests including objects using the `$not` alias I can enable it without having to also enable the `$col` alias